Providers/Modules/Plugins/Antimalware/plugin/collectdefenderinfo.rb

```diff
@@ -0,0 +1,92 @@
+require "rexml/document"
+require "cgi"
+require 'digest'
+require 'json'
+require 'date'
+require 'time'
+require 'logger'
+require_relative 'antimalwarecommon'
+
+class Defender
+
+	def self.detect()	
+		begin
+			if !File.file?('/usr/bin/mdatp')
+				return false
+			end	
+			return true
+		rescue => e					
+			return false			
+		end
+	end
+
+	def self.getprotectionstatus()		 
+		ret = {}
+		
+		defenderName = "Defender Endpoint Protection for Linux"
+		defenderVersion = "NA"
+		engineVersion = "NA"
+
+		realtimeprotection = "NA"
+		signatureVersion = "NA"
+
+		protectionStatusDetails = ""
+		protectionStatusDetailsString = ""
+
+		error = ""
+
+		($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::UnknownProtectionCode
+		($ThreatStatusRank, $ThreatStatus) = AntimalwareCommon::UnknownThreatCode
+
+		begin
+			json_data = `/usr/bin/mdatp health --output json 2>&1`
+
+			if !$?.success? || json_data.nil? || json_data.empty?				
+				error += "fail to run mdatp health cmd; "
+			end
+
+			defenderhealth = JSON.parse(json_data)
+
+			signatureStatus = defenderhealth["definitionsStatus"]["$type"]
+
+			if signatureStatus != "upToDate"
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::SignaturesOutOfDateProtectionCode
+				protectionStatusDetailsString += "Signatures out of date; "
+			end
+
+			signatureVersion = defenderhealth["definitionsVersion"]
+			defenderVersion = defenderhealth["appVersion"]
+			realtimeprotection = defenderhealth["realTimeProtectionEnabled"]["value"]
+
+			if (realtimeprotection == "NA" || !realtimeprotection)
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::NoRealTimeProtectionProtectionCode
+				protectionStatusDetailsString += "Real-time protection disabled; "
+			end
+
+			if protectionStatusDetailsString.empty? 
+				($ProtectionStatusRank, $ProtectionStatus) = AntimalwareCommon::RealTimeProtectionCode
+				protectionStatusDetailsString += "MDATP is running healthy."
+			end
+			protectionStatusDetails = protectionStatusDetailsString
+		rescue => e
+			error += "Getting exception when trying to find mcafee health info: " + e.message + " " + e.backtrace.inspect
+			ret["Error"] = error					
+		end
+
+
+		ret["ProtectionStatusRank"] = $ProtectionStatusRank
+    	ret["ProtectionStatus"] = $ProtectionStatus
+    	ret["ProtectionStatusDetails"] = protectionStatusDetails
+    	ret["DetectionId"] = SecureRandom.uuid
+    	ret["Threat"] = ""
+    	ret["ThreatStatusRank"] = $ThreatStatusRank
+    	ret["ThreatStatus"] = $ThreatStatus
+		ret["ThreatStatusDetails"] = "Threat Status is currently not supported in MDATP"
+		ret["Signature"] = (signatureVersion.nil? || signatureVersion.empty? || signatureVersion == "NA")? "Signature version not found" : signatureVersion
+    	ret["ScanDate"] = ""
```

Comment:
>"" [](http://example.com/codeflow?start=23&length=2)

empty?

Reply:
ScanDate column refers to the last known scan that the engine executed. Since MDATP does not give any API for this, I left it as empty.
I saw in other modules using "" to make sure it passes through to ODS. I can try removing it, but I also left it there for future look overs in this code to say, "No, we don't collect scan dates"